src/infrastructure/delivery/provider/entities.ts

```diff
@@ -0,0 +1,82 @@
+export type Email = {
+    address: string
+    name: string
+}
+
+export type TypeMessage = {
```

Comment:
I think _MessageType_ sounds better, I feel the name is long, we can discuss the convention for the type and interface names to apply on all the codebase